server/fleet/app.go

```diff
@@ -571,6 +571,13 @@ func (c *AppConfig) Copy() *AppConfig {
 			clone.Integrations.Zendesk[i] = &zd
 		}
 	}
+	if len(c.Integrations.GoogleCalendar) > 0 {
+		clone.Integrations.GoogleCalendar = make([]*GoogleCalendarIntegration, len(c.Integrations.GoogleCalendar))
+		for i, g := range c.Integrations.GoogleCalendar {
+			gc := *g
+			clone.Integrations.GoogleCalendar[i] = &gc
+		}
+	}
```

Comment:
Great that you caught this issue. How did you catch it?

Reply:
I was getting the `MaskedPassword` as private key.

Reply:
Note that you would get a CI failure if a new field was added to a struct that requires cloning (https://github.com/fleetdm/fleet/blob/61544f4beacb21570ef4cffbfa29c199fb8b491d/.github/workflows/golangci-lint.yml#L66-L68). This failure is meant as a reminder to add the cloning logic if necessary for that new field/type.